iothub/device/src/Pipeline/MqttTransportHandler.cs

```diff
@@ -1181,10 +1178,10 @@ private void HandleTwinResponse(MqttApplicationMessageReceivedEventArgs received
             {
                 byte[] payloadBytes = receivedEventArgs.ApplicationMessage.Payload ?? Array.Empty<byte>();
 
-                if (_getTwinResponseCompletions.TryRemove(receivedRequestId, out TaskCompletionSource<GetTwinResponse> getTwinCompletion))
+                if (_pendingTwinOperations.TryRemove(receivedRequestId, out PendingMqttTwinOperation patchTwinOperation))
```

Comment:
This path was semi broken, written part as "get twin" and other part as "patch twin". I updated it to be patch twin. The difference is that the version is returned.